### PR TITLE
Python bindings: GIL control

### PR DESCRIPTION
**Pull request by Yarikk26**
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] Pull request title reflects its content
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Description:
Python bindings strive to keep GIL released during C++ calls.

### CODE REVIEW

File: bindings/python/core_objects/include/py_core_objects/py_variant_extractor.h [hunk 1]
```diff
@@ -169,6 +170,7 @@ template <typename DaqType,
           typename Variant>
 daq::ObjectPtr<std::remove_pointer_t<DaqType>> getVariantValue(Variant& v)
 {
+    py::gil_scoped_acquire acquire;
```
Comment (dejancrn) [inline bindings/python/core_objects/include/py_core_objects/py_variant_extractor.h:173]: As discussed, this should be investigated why this line is neccessary.

**Review by dejancrn — CHANGES_REQUESTED** [review]
Looks good, however, I would like to have an explanation for the reason of the crash without GIL acquire in getVariantValue call.

File: bindings/python/core_objects/include/py_core_objects/py_variant_extractor.h (hunk 1 above)
Comment (Yarikk26) [reply bindings/python/core_objects/include/py_core_objects/py_variant_extractor.h]: It's acquiring GIL only for python types now

**Comment by Yarikk26** [comment]
I've made some changes in **RTGen** to generate a little more neat code. I've also abandoned the idea with `py::call_guard`, because it needs some nested scopes and code reindenting for `def_property`

File: bindings/python/core_objects/include/py_core_objects/py_variant_extractor.h [hunk 2]
```diff
@@ -195,6 +196,9 @@ daq::ObjectPtr<std::remove_pointer_t<DaqType>> getVariantValue(Variant& v)
     }
     else if (auto native = std::get_if<NativeType>(&v))
     {
+        std::optional<py::gil_scoped_acquire> acquire; 
+        if (std::is_base_of_v<py::handle, NativeType>) acquire.emplace();
```
Comment (dejancrn) [inline bindings/python/core_objects/include/py_core_objects/py_variant_extractor.h:200]: Use "if constexpr ()" here. It can be done in compile time.

**Review by dejancrn — APPROVED** [review]

**Pull request merged**